### PR TITLE
Added CI

### PR DESCRIPTION
**Pull request by cliedeman**
Working build - https://github.com/cliedeman/relay-hooks/runs/513078402

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - uses: actions/checkout@v1
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: build
+        run: |
+          npm ci
+          npm run format
```
Comment (morrys) [inline .github/workflows/build.yaml]: I don't really like the idea of the format in a CI.
I prefer that here only be verified that it is formatted correctly

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: Build
+on: [push]
+
+jobs:
```
Comment (morrys) [inline .github/workflows/build.yaml]: How about dividing the steps into more Jobs?
  * build
  * eslint
  * test (here at the moment it's also necessary to build)

File: .github/workflows/build.yaml (hunk 1 above)
Comment (cliedeman) [reply .github/workflows/build.yaml]: Oops. Thanks will fix

File: .github/workflows/build.yaml (hunk 2 above)
Comment (cliedeman) [reply .github/workflows/build.yaml]: Can do

**Comment by cliedeman** [comment]
@morrys updated.

https://github.com/cliedeman/relay-hooks/runs/515955930

File: package.json [hunk 3]
```diff
@@ -25,6 +25,7 @@
     "build": "npm run clean && tsc && npm run test",
```
Comment (morrys) [inline package.json]: hi @cliedeman,  for a better optimization I would ask you to modify the scripts in the package.json like this:
```
"scripts": {
    "clean": "rimraf lib",
    "compile": "npm run clean && tsc",
    "build": "npm run compile && npm run test",
    "test": "cross-env NODE_ENV=test jest --coverage",
    "format": "prettier --write \"src/**/*.{j,t}s*\"",
    "eslint": "eslint ./src --ext .js,.jsx,.ts,.tsx",
    "prepublishOnly": "npm run build"
 },
```

And to change the step and the **build** command to **compile**.

These changes avoid _compile_ and _test_ in the **install dependencies** step and _tests_ in the **build** step.

what do you think about it? :)

Comment (cliedeman) [reply package.json]: No strong preference. Done

**Pull request merged**